app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -30,6 +33,38 @@ class WallpaperManager(private val settings: Settings) {
             wallpaperContainer.setBackgroundResource(newWallpaper.drawable)
         }
         currentWallpaper = newWallpaper
+
+        adjustTheme(wallpaperContainer.context)
+    }
+
+    private fun adjustTheme(context: Context) {
+        val mode = if (currentWallpaper != Wallpaper.NONE) {
+            if (currentWallpaper.isDark) {
+                updateThemePreference(useDarkTheme = true)
+                AppCompatDelegate.MODE_NIGHT_YES
+            } else {
+                updateThemePreference(useLightTheme = true)
+                AppCompatDelegate.MODE_NIGHT_NO
+            }
+        } else {
+            updateThemePreference(followDeviceTheme = false)
+            AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+        }
+
+        if (AppCompatDelegate.getDefaultNightMode() != mode) {
+            AppCompatDelegate.setDefaultNightMode(mode)
+            context.asActivity()?.recreate()
+        }
```

Comment:
Could this cause the `settings.shouldFollowDeviceTheme` and the `AppCompatDelegate` property to be out of sync? The theme preference would be updated with `followDeviceTheme = false`, but `AppCompatDelegate.setDefaultNightMode` will be called with `MODE_NIGHT_FOLLOW_SYSTEM`

Reply:
This is just visual improvement to avoid recreating the fragment when it's not needed and avoid flashes, as we are only recreating the fragment/`setDefaultNightMode` when the mode is different of the actual one, I don't think it will get out sync, if it's the same we don't need to update it again and cause a flash.

Reply:
> The theme preference would be updated with followDeviceTheme = false, but AppCompatDelegate.setDefaultNightMode will be called with MODE_NIGHT_FOLLOW_SYSTEM

In the above example, we are only going to update `setDefaultNightMode`/`asActivity()?.recreate()` if the actual mode (`AppCompatDelegate.getDefaultNightMode()) is different of the actual one, it's the same there is not need for updating it.

Imagine that we have three wallpapers in the following order `[wallpaper 1 is dark,  wallpaper 2 is dark, wallpaper 3 is light]`, if you are on light mode and you tap the logo three times:

1. Wallpaper 1 will be applied and we will call `setDefaultNightMode` to set dark mode.
2. Wallpaper 2 will be applied and we will **NOT call** `setDefaultNightMode` as we are already on dark mode.
3. Wallpaper 3 will be applied and we will call `setDefaultNightMode` to set light mode.

Reply:
What happens when we wrap back around? Is the default wallpaper now light, even if the user originally was following a dark system theme?

Reply:
OK, I see what do you mean.  I updated  `updateThemePreference` to `updateThemePreference(followDeviceTheme = true)`